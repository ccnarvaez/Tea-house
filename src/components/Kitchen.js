export const Kitchen = () => {

    return(

        <>
            <h1> Esta es la cocina! </h1>
        </>

    )
}