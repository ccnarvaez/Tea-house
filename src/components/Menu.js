import { data } from "../data/menu";
import { MenuGrid } from "../views/MenuGrid";
import { useState } from "react";




export const Menu = () => {


    const [order, setOrder] = useState([]);
    
   

    return(
    <>

        <div className = "wrapper"> 
            { data.map( elem => (

                <MenuGrid id= {elem.id} item = { elem.item } price = { elem.price } img = { elem.img} />

            ))};
        </div> 

        <div> { order } </div>
          
    </>
    )
};

