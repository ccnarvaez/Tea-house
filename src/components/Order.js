import { MenuGrid } from "../views/MenuGrid";



export const Order = () => {
    const date = new Date();
        const whatTime = date.toLocaleTimeString();

    

    return(

        <>
            <div> {whatTime} </div>
            <p> <MenuGrid /> </p>
        </>
    )
}