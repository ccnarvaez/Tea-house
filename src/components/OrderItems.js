let order =[];

export const OrderItems = ( e )=> {
    let index = order.findIndex( element => element.id === e.target.dataset.id );
    if ( index === -1){
        let item = {
            id: e.target.dataset.id,
            item: e.target.dataset.item,
            price: e.target.dataset.price,
            quantity: 1,
        };

        order.push( item );
    }
    else{            
        order[index].quantity += 1;
    }

    console.log( order )

};