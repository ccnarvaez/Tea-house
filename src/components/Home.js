import { useState } from "react";
import { Menu } from "../components/Menu";



export const Home = () => {
    const [  menu, setMenu ] = useState([]);

    const handleMenu = () => {

        setMenu( () => {
            return (
                <div> <Menu /> </div>
            )   
        })
    };

    return (
        <>
            <h1 className = "animate__animated animate__rubberBand"> Tea house </h1>
            <h2 className = "animate__animated animate__rubberBand" > Elige lo que quieres hacer </h2>
            
            <div className="btn-group" role="group" aria-label="Basic radio toggle button group">
                <button onClick = { handleMenu } className="btn btn-success btn-lg"> Crear orden </button>
                <button className="btn btn-warning btn-lg"> Cocina </button>
            </div>
            <hr></hr>
            
            <div> { menu } </div>
        
          
        </>

    )
}