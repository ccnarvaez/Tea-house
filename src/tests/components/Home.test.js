import { shallow } from "enzyme";
import React from "react";
import { Home } from "../../components/Home";

describe( 'Testing <Home /> component', () => {

    const wrapper = shallow( 
        <Home />
    );
    test('Should render properly', () => {
        expect( wrapper ).toMatchSnapshot();
    })
})