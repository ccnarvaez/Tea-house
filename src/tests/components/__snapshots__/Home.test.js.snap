// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Home /> component Should render properly 1`] = `ShallowWrapper {}`;
